fhir-search/src/main/java/com/ibm/fhir/search/sort/Sort.java

```diff
@@ -157,13 +166,18 @@ public boolean isUndefinedOrLenient(String resourceTypeName, String sortParmCode
      * @param resourceTypeName one of the FHIR Resources, and must be a non-null
      *                         value
      * @param code             the code to check.
+     * @param context          the context
      * @throws FHIRSearchException
      */
-    public void checkSystemLevel(String resourceTypeName, String code) throws FHIRSearchException {
+    public void checkSystemLevel(String resourceTypeName, String code, FHIRSearchContext context) throws FHIRSearchException {
         if ("Resource".compareTo(resourceTypeName) == 0
                 && !SYSTEM_LEVEL_SORT_PARAMETER_NAMES.contains(code)) {
-            throw SearchExceptionUtil.buildNewInvalidSearchException(
-                    buildNewInvalidSearchExceptionMessage(code));
+            String msg = buildNewInvalidSearchExceptionMessage(code);
+            if (context.isLenient()) {
+                log.fine(msg);
```

Comment:
isLoggable

Reply:
The log.fine(...) method internally does an isLoggable check.

Reply:
that's correct, as a general pattern we've followed isloggable, more specifically is the msg is annotated or constructed.
it'd be decent to be defensive and not build the msg unless we need to, and log only as needed.

Reply:
Yeah, I agree.  In this case, the msg always needs to be constructed to be added to the OutcomeIssues and/or the thrown exception, so I didn't add the isLoggable check.

Reply:
makes sense